### DISCUSSION TITLE
Mutants inside ignored methods - Bug or Normal behaviour ?

### DISCUSSION THREAD

**Discussion by GaetBaka** [stryker-mutator/stryker-net#1841] [Q&A] [answered]
Hello,
As a stryker newbie, I don't want to create a bug if that's a normal behaviour.

I have encountered two issues while implementing stryker 0.22.10 for my .net core project, that are kinda similar : mutants inside ignored methods.

Config-
Inside of my stryker-config files, I have ignored the following methods, used for tracing the app through logs & metrics : 
 "ignore-methods": [ "LogInformation", "IncrementCounter",  "LogReplicatedOfferAndUpdateMetrics", "WriteHistogram" ]

Issue 1- 
I've been forced to add WriteHistogram as an ignored method despite having ignored LogReplicatedOfferAndUpdateMetrics. 
The only time it is called is inside this method, but it still created a mutant thus resulting in a zombie, even if I wanted to ignore LogReplicatedOfferAndUpdateMetrics.

![image](https://user-images.githubusercontent.com/81621060/146579272-4cb30957-a224-47d2-8d73-6e177e4c409b.png)
![image](https://user-images.githubusercontent.com/81621060/146579577-07d53b8e-6996-4bab-8955-cb8ec49891df.png)

Issue 2-
Not inside the ignored method this time, but as an attribute of it : 
I have ignored the method LogInformation, but it still created a mutant because there is a method string.join inside of it. 
I don't want to ignore string.join methods because it might impact the rest of the code.
![image](https://user-images.githubusercontent.com/81621060/146579745-29091938-2155-470d-9302-22b98a6ef8a8.png)

So I ask the question : Is this normal behaviour ? Do I have to go one step further in ignoring the methods, by adding an attribute somewhere, so that it also ignores everything inside of them ?

Thanks for reading, and I hope you'll have some answers for me, or that I helped contributing in this project :]

Regards,
Gaétan.

**Comment by rouke-broersma** [comment, marked as answer]
Could you update to the latest version? This should be possible.

**Reply by GaetBaka** [reply]
Yea I just realized we were so far behind, because using .net core 3.1
The disable option been added in version 5.4 so it's our fault :D
Sorry for that

**Reply by rouke-broersma** [reply]
You can use stryker on .netcore 3.1 as long as you have the .net5 runtime installed on your systems. With dotnet core it is possible to have them side-by-side installed without affecting each other.
